src/chains/ethereum/options/tests/logging-options.test.ts

```diff
@@ -0,0 +1,203 @@
+          options.logging.logger.log(message);
+          assert.strictEqual(spy.withArgs(message).callCount, 1);
+        });
+
+        it("disables the logger when the quiet flag is used", () => {
+          const message = "message";
+          const options = EthereumOptionsConfig.normalize({
+            logging: { quiet: true }
+          });
+          options.logging.logger.log(message);
+          assert.strictEqual(spy.withArgs(message).callCount, 0);
+        });
+
+        it("calls the provided logger when quiet flag is used", () => {
+          const logLines: string[][] = [];
+          const options = EthereumOptionsConfig.normalize({
+            logging: {
+              quiet: true,
+              logger: {
+                log: (message: any, ...params: any[]) =>
+                  logLines.push([message, ...params])
+              }
+            }
+          });
+
+          options.logging.logger.log("message", "param1", "param2");
+
+          assert.deepStrictEqual(logLines, [["message", "param1", "param2"]]);
+        });
+      });
+
+      describe("file", () => {
+        it("resolves a file path to descriptor", async () => {
+          const options = EthereumOptionsConfig.normalize({
+            logging: { file: validFilePath }
+          });
+          try {
+            assert(typeof options.logging.file === "number");
+            assert.doesNotThrow(
+              () => closeSync(options.logging.file),
+              "File descriptor not valid"
+            );
+          } finally {
+            await unlink(validFilePath);
+          }
+        });
+
+        it("resolves a file path as Buffer to descriptor", async () => {
+          const options = EthereumOptionsConfig.normalize({
+            logging: { file: Buffer.from(validFilePath, "utf8") }
+          });
+          try {
+            assert(typeof options.logging.file === "number");
+            assert.doesNotThrow(
+              () => closeSync(options.logging.file),
+              "File descriptor not valid"
+            );
+          } finally {
+            await unlink(validFilePath);
+          }
+        });
+
+        it("resolves a file URL as Buffer to descriptor", async () => {
+          const options = EthereumOptionsConfig.normalize({
+            logging: { file: new URL(`file://${validFilePath}`) }
+          });
+          try {
+            assert(typeof options.logging.file === "number");
+            assert.doesNotThrow(
+              () => closeSync(options.logging.file),
+              "File descriptor not valid"
+            );
+          } finally {
+            await unlink(validFilePath);
+          }
+        });
+
+        it("fails if the process doesn't have write access to the file path provided", async () => {
+          const file = resolve("./eperm-file.log");
+          try {
+            const handle = await open(file, "w");
+            // set no permissions on the file
+            await handle.chmod(0);
```

Comment:
How does `handle.chmod()` behave on Windows? is it necessary or does the test work because the handle is closed in the next line?

Reply:
`chmod` with `0` will add the `r` (readonly) flag, causing writes to the file to fail:
```
> fs.chmodSync("./testfile.log", 0);
> fs.appendFileSync("./testfile.log", "hi");
Uncaught Error: EPERM: operation not permitted, open './testfile.log'
    at Object.openSync (fs.js:498:3)
    at Object.writeFileSync (fs.js:1529:35) {
  errno: -4048,
  syscall: 'open',
  code: 'EPERM',
  path: './testfile.log'
}

...

-ar---         3/13/2023  12:33 PM             10 testfile.log
```